WcaOnRails/app/models/competition_series.rb

```diff
@@ -129,4 +129,8 @@ def self.wcif_to_attributes(wcif)
     # reset them so that upon the next read they will be fetched based on what's just been written.
     @competition_ids = nil
   end
+
+  def public_competitions
+    self.competitions.filter(&:showAtAll?)
```

Comment:
Are you sure this uses the ORM query interface directly? I _think_ (I am genuinely not sure) `filter` first loads all competitions into memory and then throws away the hidden ones. With an approach like `self.competitions.where(showAtAll: false)` we could make sure that the Rails ActiveRecord ORM only loads those competitions into memory that are public in the first place.

Reply:
Ah good point. Thanks, Gregor!